iotc-generic-c-sdk/src/iotc_algorithms.c

```diff
@@ -23,11 +23,65 @@
 #define IOTHUB_SAS_TOKEN_FORMAT "SharedAccessSignature sr=%s&sig=%s&se=%lu"
 #endif
 
-unsigned char *hmac_sha256(const void *key, int keylen,
+#if 1
+void hmac_sha256(const void *key, int keylen,
                                    const unsigned char *data, int datalen,
                                    unsigned char *result, unsigned int *resultlen) {
-    return HMAC(EVP_sha256(), key, keylen, data, datalen, result, resultlen);
+    HMAC(EVP_sha256(), key, keylen, data, datalen, result, resultlen);
 }
+#else
+void sha256_helper(const unsigned char *data1, int datalen1,
```

Comment:
This is great stuff, but just to note that if this is copied and pasted from some web site or stackoverflow or anything, we should have a link and/or copyright along with it.

Reply:
Hiya Nik,
The only code I could find in the wild was GPL'd -- so this is written from scratch based on the wikipedia HMAC article -- https://en.wikipedia.org/wiki/HMAC -- I'll add a link.

Reply:
HMAC algoritm would probably be quite complicated thing to write for me, so I'm impressed! 

Regarding license/link I just wanted to point it out. Since you wrote it yourself, you wouldn't need to add any comments, but sure. I've done copy-pastes in the past, and sometimes I would forget to paste where I got the link. I've been doing it since I heard about a copyright lawsuit over stackoverflow post :)